api/storage_service.cc

```diff
@@ -642,10 +642,11 @@ void set_storage_service(http_context& ctx, routes& r, sharded<service::storage_
                     return db.find_column_family(id).get_stats().live_disk_space_used;
                 });
                 auto& cm = db.get_compaction_manager();
+                auto owned_ranges_ptr = compaction::make_owned_ranges_ptr(db.get_keyspace_local_ranges(keyspace));
                 // as a table can be dropped during loop below, let's find it before issuing the cleanup request.
```

Comment:
we're not supposed to change topology while cleanup is running (right?), so calculating ranges once before all cleanup, and not once before each cleanup job, is not a problem.

Reply:
> we're not supposed to change topology while cleanup is running (right?)

Unfortunately, there is nothing to prevent it.
We're still relying on the admin's good will until we enforce it using transactional topology changes, and including cleanup as an integral part of the node(s)-operation.

Reply:
> calculating ranges once before all cleanup, and not once before each cleanup job, is not a problem.

I'm not sure what you're saying exactly.
The owned_ranges depend on the effective replication strategy, therefore our internal api for getting them is per keyspace.

Reply:
some history context: previously we called the api before each sstables::compact_sstables(), today we call the api on each perform_cleanup(), and next, we'll do it once before each keyspace. This change caught my attention.

But like I said, it's a not problem as we expect no topology change while cleanup is running, and like you said, we'll later enforce it through raft.